database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -0,0 +1,438 @@
+
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class SqlDatabaseService implements DatabaseService {
+    private final Settings settings = new Settings();
+    private static final String dbURL = "jdbc:mysql://localhost:3306/entities";
+    private static final String dbUserName = "bd_user";
+    private static final String dbPassword = "Qwerty!1";
+
+    private static final Logger LOG = LoggerFactory.getLogger(SqlDatabaseService.class);
+    static final String UNABLE_CREATE_TABLE = "Unable to create table. Please check if it already exists";
+    static final String UNABLE_DELETE_TABLE = "Unable to delete table. Please check if table does not exist";
+    static final String TABLE_NOT_EXIST = "Table does not exist";
+    static final String UNABLE_ADD_NEW_RECORD = "Unable to add new record to table";
+    static final String UNABLE_DELETE_RECORD = "Unable to delete record from table";
+    static final String UNABLE_DELETE_ALL_RECORDS = "Unable to delete all records from table";
+    static final String ENTITY_IS_NOT_FOUND = "Entity with provided Id is not found";
+    static final String ID_PROVIDED_MANUALLY = "User cannot provide id manually. Ids are filled automatically";
+    static final String INVALID_PARAMETER_VALUE = "Invalid parameter value. " +
+            "Limit value should be in(0..{MAX_LIMIT_VALUE}), offset value should be >= 0";
+
+    @Override
+    public boolean createTable(Class<? extends BaseEntity> entityClass) {
+        String tableName = entityClass.getSimpleName();
+        StringBuilder createTableSQL = new StringBuilder("CREATE TABLE " + tableName + " (");
+
+        createTableSQL.append("id INT AUTO_INCREMENT PRIMARY KEY, ");
+
+        List<Field> fields = getAllFields(entityClass);
+
+        for (Field field : fields) {
+            String fieldName = field.getName();
+            if (fieldName.equals("id")) {
+                continue;
+            }
+            String fieldType = getSQLType(field.getType());
+            createTableSQL.append(fieldName).append(" ").append(fieldType).append(", ");
+        }
+
+        createTableSQL.setLength(createTableSQL.length() - 2);
+        createTableSQL.append(")");
+
+        LOG.info("Executing SQL: {}", createTableSQL);
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement()) {
+            statement.executeUpdate(createTableSQL.toString());
+            return true;
+        } catch (SQLException e) {
+            LOG.error(UNABLE_CREATE_TABLE + ": {}", tableName);
+            throw new CreationDatabaseException(UNABLE_CREATE_TABLE + ": " + e.getMessage());
```

Comment:
Please add table name here. You can prepare message and pass it to `log` and to `throw`.